addons/portal/controllers/portal.py

```diff
@@ -550,6 +556,21 @@ def _create_or_update_address(
                 # The `phone_validation` module is installed.
                 partner_sudo._onchange_phone_validation()
 
+        if (
+            'company_name' in address_values
+            and not partner_sudo.is_company
+            and partner_sudo.type == 'contact'
+        ):
```

Comment:
this if only makes sense when we update an address, not when we create a new one, it can be inside `elif not self._are_same_addresses`.

```suggestion
        if (
            'company_name' in address_values
            and partner_sudo.commercial_partner_id != partner_sudo
            and partner_sudo.commercial_partner_id.is_company
        ):
```
would be more generic than checking the type and whether the partner is a company.